pkg/sql/truncate.go

```diff
@@ -413,107 +415,90 @@ func reassignComment(
 		}
 	}
 
-	for i := range oldTableDesc.Columns {
-		id := oldTableDesc.Columns[i].ID
-		err = reassignColumnComment(ctx, p, oldTableDesc.ID, newTableDesc.ID, id)
-		if err != nil {
-			return err
-		}
+	if err := reassignAllColumnComments(ctx, p, oldTableDesc.ID, newTableDesc.ID, oldTableDesc.Columns); err != nil {
+		return err
 	}
 
-	for _, indexDesc := range oldTableDesc.Indexes {
-		err = reassignIndexComment(ctx, p, oldTableDesc.ID, newTableDesc.ID, indexDesc.ID)
-		if err != nil {
-			return err
-		}
+	if err := reassignAllIndexComments(ctx, p, oldTableDesc.ID, newTableDesc.ID, oldTableDesc.Indexes); err != nil {
+		return err
 	}
 
 	return nil
 }
 
-// reassignColumnComment reassign comment on column.
-func reassignColumnComment(
-	ctx context.Context, p *planner, oldID sqlbase.ID, newID sqlbase.ID, columnID sqlbase.ColumnID,
+// reassignAllColumnComments moves all column comments from oldID to newID.
+func reassignAllColumnComments(
+	ctx context.Context,
+	p *planner,
+	oldID sqlbase.ID,
+	newID sqlbase.ID,
+	columns []sqlbase.ColumnDescriptor,
 ) error {
-	comment, err := p.ExtendedEvalContext().ExecCfg.InternalExecutor.QueryRowEx(
+	if len(columns) == 0 {
+		return nil
+	}
+	var in strings.Builder
+	in.WriteString("(")
+	for i := range columns {
+		if i > 0 {
+			in.WriteString(", ")
+		}
+		in.WriteString(fmt.Sprintf("%d", columns[i].ID))
+	}
+	in.WriteString(")")
+	inString := in.String()
+
+	_, err := p.ExtendedEvalContext().ExecCfg.InternalExecutor.ExecEx(
 		ctx,
-		"select-column-comment",
+		"update-column-comments",
 		p.txn,
 		sqlbase.InternalExecutorSessionDataOverride{User: security.RootUser},
-		`SELECT comment FROM system.comments WHERE type=$1 AND object_id=$2 AND sub_id=$3`,
-		keys.ColumnCommentType,
+		fmt.Sprintf(
+			`UPDATE system.comments SET object_id=$1 WHERE object_id=$2 AND type=$3 AND sub_id IN %s`,
```

Comment:
Sorry to keep going back and forth on this but can't you just do

`update system.comments set object_id=<new> where object_id=<old>` and be done with it?